src/main/java/sirius/biz/analytics/charts/explorer/EventTimeSeriesComputer.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.analytics.charts.explorer;
+
+import sirius.biz.analytics.events.Event;
+import sirius.biz.analytics.events.EventRecorder;
+import sirius.db.jdbc.OMA;
+import sirius.db.jdbc.Row;
+import sirius.db.jdbc.SmartQuery;
+import sirius.kernel.commons.Limit;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Tuple;
+import sirius.kernel.di.std.Part;
+
+import javax.annotation.Nullable;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * Provides a time series computer which operates on {@link Event events}.
+ * <p>
+ * Note that this is intended for straight forward queries which need to be adaptive
+ * (like {@link sirius.biz.tycho.updates.NumberOfUpdateClicksChart} or which are re-usable like
+ * {@link UserAgentsTimeSeriesComputer}). If a query grows and becomes too complex or isn't buildable using
+ * {@link SmartQuery} anyway, a custom computer should be implements (most probably as anonymous lambda in the
+ * chart itself), which directly builds the query as readable and maintainable SQL string and collects its result.
+ *
+ * @param <O> the types of objects on which this computer operates
+ * @param <E> the events being processed
+ */
+public class EventTimeSeriesComputer<O, E extends Event> implements TimeSeriesComputer<O> {
+
+    /**
+     * Provides a common aggregation which counts all events (<tt>count(*)</tt>).
+     */
+    public static final String AGGREGATION_EXPRESSION_COUNT = "count(*)";
+
+    /**
+     * Counts all events containing a {@link sirius.biz.analytics.events.UserData}, where the
+     * {@link sirius.biz.analytics.events.UserData#USER_ID} is filled.
+     */
+    public static final String AGGREGATION_EXPRESSION_COUNT_LOGGED_IN = "countIf(userData_userId IS NOT NULL)";
+
+    /**
+     * Provides a proper label for {@link #AGGREGATION_EXPRESSION_COUNT_LOGGED_IN}.
+     */
+    public static final String LABEL_LOGGED_IN = "$EventTimeSeriesComputer.loggedIn";
```

Comment:
Move down to the other label?

Reply:
currently the constants are grouped logically (expression +label; expression+label)....at least, that was the idea...